inbox.datetime.format does not support date only values

**Describe the bug**
A clear and concise description of what you see:
When configuring the inbox to with only a date (and no time) to be matched, you'll see a `DateTimeParseException` when putting video files into the inbox matching that pattern.

**To Reproduce**
Steps to reproduce the behavior:
1. Configure your inbox:
2. `inbox.datetime.format = yyyy-MM-dd`
3. `inbox.metadata.regex = (?<created>....-..-..)[ _](?<spatial>[^ ]+) (?<title>.+)\\.(?:mp4|mov|MP4|MOV)`.
4. Drop a file named `2021-12-10 hs1 Biology 101.mp4` into the inbox.
5. See admin error log:

```
java.time.format.DateTimeParseException: Text '2022-05-24' could not be parsed: Unable to obtain LocalDateTime from TemporalAccessor: {},ISO resolved to 2022-05-24 of type java.time.format.Parsed
	at java.time.format.DateTimeFormatter.createError(DateTimeFormatter.java:2017) ~[?:?]
	at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1952) ~[?:?]
	at java.time.LocalDateTime.parse(LocalDateTime.java:492) ~[?:?]
	at org.opencastproject.ingest.scanner.Ingestor$RetriableIngestJob.lambda$call$2(Ingestor.java:198) [!/:?]
	at org.opencastproject.security.util.SecurityContext.runInContext(SecurityContext.java:58) [!/:?]
	at org.opencastproject.ingest.scanner.Ingestor$RetriableIngestJob.call(Ingestor.java:158) [!/:?]
	at org.opencastproject.ingest.scanner.Ingestor$RetriableIngestJob.call(Ingestor.java:131) [!/:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: java.time.DateTimeException: Unable to obtain LocalDateTime from TemporalAccessor: {},ISO resolved to 2022-05-24 of type java.time.format.Parsed
	at java.time.LocalDateTime.from(LocalDateTime.java:461) ~[?:?]
	at java.time.format.Parsed.query(Parsed.java:235) ~[?:?]
	at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1948) ~[?:?]
	... 11 more
Caused by: java.time.DateTimeException: Unable to obtain LocalTime from TemporalAccessor: {},ISO resolved to 2022-05-24 of type java.time.format.Parsed
	at java.time.LocalTime.from(LocalTime.java:431) ~[?:?]
	at java.time.LocalDateTime.from(LocalDateTime.java:457) ~[?:?]
	at java.time.format.Parsed.query(Parsed.java:235) ~[?:?]
	at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1948) ~[?:?]
	... 11 more
```

It would be helpful if times could be omitted.

**Environment Information**
 - OS: based on Opencast Docker image
 - Opencast version with minor release 11.7

--- a/modules/ingest-service-impl/src/main/java/org/opencastproject/ingest/scanner/InboxScannerService.java
+++ b/modules/ingest-service-impl/src/main/java/org/opencastproject/ingest/scanner/InboxScannerService.java
@@ -57,6 +57,8 @@ import org.slf4j.LoggerFactory;
 import java.io.File;
 import java.io.IOException;
 import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.temporal.ChronoField;
 import java.util.Dictionary;
 import java.util.Enumeration;
 import java.util.HashMap;
@@ -195,7 +197,11 @@ public class InboxScannerService implements ArtifactInstaller, ManagedService {
             .map(Pattern::compile);
     var dateFormatter = Optional.ofNullable(properties.get(INBOX_DATETIME_FORMAT))
             .map(Objects::toString)
-            .map(DateTimeFormatter::ofPattern)
+            .map(s -> new DateTimeFormatterBuilder().appendPattern(s)
+                    .parseDefaulting(ChronoField.HOUR_OF_DAY, 0)
+                    .parseDefaulting(ChronoField.MINUTE_OF_HOUR, 0)
+                    .parseDefaulting(ChronoField.SECOND_OF_MINUTE, 0)
+                    .toFormatter())
             .orElse(DateTimeFormatter.ISO_DATE_TIME);
     var ffprobe = BooleanUtils.toBoolean((String) properties.get(INBOX_METADATA_FFPROBE))
             ? Objects.toString(cc.getBundleContext().getProperty(FFPROBE_BINARY_CONFIG), FFPROBE_BINARY_DEFAULT)

